### PR TITLE
ci: create update index.yaml GitHub action (glasskube/glasskube#69)

### PR DESCRIPTION
**Pull request by pmig**
resolves glasskube/glasskube#69

### CODE REVIEW

File: .github/workflows/update-index.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  update-index:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+        with:
+          fetch-depth: 0
+      - name: Generate index.yaml
+        uses: mikefarah/yq@master
```
Comment (kosmoz) [inline .github/workflows/update-index.yaml:16]:
```suggestion
        uses: mikefarah/yq@dd648994340a5d03225d97abf19c9bf1086c3f07 # v4.40.5
```

File: .github/workflows/update-index.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        with:
+          cmd: yq ea '[.]' packages/**/package.yaml | yq '... comments=""' | yq '{"packages":.}' > packages/index.yaml
+      - name: Commit index.yaml
```
Comment (kosmoz) [inline .github/workflows/update-index.yaml:19]: why not simply use git?

https://stackoverflow.com/questions/57921401/push-to-origin-from-github-action/58393457#58393457

**Review by kosmoz — CHANGES_REQUESTED** [review]

File: .github/workflows/update-index.yaml (hunk 2 above)
Comment (kosmoz) [reply .github/workflows/update-index.yaml:19]: Now that I think about it… would it make sense for the Workflow to create a PR instead? That way we could prevent having a broken index file caused by a bad workflow.

Comment (pmig) [reply .github/workflows/update-index.yaml]: migrated to git and and will now create a pr

**Review by kosmoz — APPROVED** [review]
Looks good :+1: Does the last step also work correctly if there are no changes? Or does it create an empty PR. In the end, both would be fine I guess.

**Comment by pmig** [comment]
> Looks good 👍 Does the last step also work correctly if there are no changes? Or does it create an empty PR. In the end, both would be fine I guess.

We would need to find out.

**Pull request merged**